### PR TITLE
Refactor empty separator case for String#split

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by MaxGraey** [comment]
Also synced

File: std/assembly/string.ts [hunk 1]
```diff
@@ -407,7 +407,11 @@ export class String {
   split(separator: String = null, limit: i32 = i32.MAX_VALUE): String[] {
     assert(this !== null);
     if (!limit) return new Array<String>();
-    if (separator === null) return <String[]>[this];
+    if (separator === null) {
+      let result = new Array<String>(1);
+      unchecked(result[0] = this);
```
Comment (dcodeIO) [inline std/assembly/string.ts]: Is this only about the unchecked operation here, or was there also a bug?

Comment (MaxGraey) [reply std/assembly/string.ts]: As you remember arrays literals always store in data section instead alloc in heap. Or this was fixed already?

Comment (dcodeIO) [reply std/assembly/string.ts]: If the elements aren't all static, which is the case here, it should (have) become a normal instantiation. There' s still that issue where const arrays within a function can be mutated before the next return, though, but it shouldn't be related I think. That's the one you are referring to?

Comment (MaxGraey) [reply std/assembly/string.ts]: yeah. btw it will be nice fixed this asap. Currently I close this because this different case don't need workaround

Comment (dcodeIO) [reply std/assembly/string.ts]: I see, agree. One way to solve this other issue could be to allocate the initial buffer in static memory only, and memcpy it to the array's backing buffer on use, so it always becomes a new ref.

**Pull request closed without merging**